_playwright-tests/Integration/CanUpdateSystemWithTemplate.spec.ts

```diff
@@ -0,0 +1,156 @@
+      await page.getByRole('button', { name: 'Next', exact: true }).click();
+      await page.getByRole('button', { name: 'Next', exact: true }).click();
+      await page.getByText('Use up to specific date', { exact: true }).click();
+      await page.getByPlaceholder('YYYY-MM-DD', { exact: true }).fill('2021-05-17'); // Older than any snapshot date
+      await page.getByRole('button', { name: 'Next', exact: true }).click();
+      await page.getByText('add template modal', { exact: true });
+      await page.getByPlaceholder('Enter name').fill(`${templateName}`);
+      await page.getByPlaceholder('Description').fill('Template test');
+      await page.getByRole('button', { name: 'Next', exact: true }).click();
+      await page.getByRole('button', { name: 'Create other options' }).click();
+      await page.getByText('Create template only', { exact: true }).click();
+      const rowTemplate = await getRowByNameOrUrl(page, `${templateName}`);
+      await expect(rowTemplate.getByText('Valid')).toBeVisible({ timeout: 60000 });
+    });
+
+    await test.step('Create RHSM client and register the template', async () => {
+      // change the test timeout as registering a client can be slow
+      test.setTimeout(10 * 60 * 1000); // ten minutes
+
+      // Start the rhel9 container
+      await client.Boot('rhel9');
+
+      // Register, overriding the default key and org
+      const reg = await client.RegisterRHC(
+        process.env.ACTIVATION_KEY_1,
+        process.env.ORG_ID_1,
+        templateName,
+      );
+      if (reg?.exitCode != 0) {
+        console.log(reg?.stdout);
+        console.log(reg?.stderr);
+      }
+      expect(reg?.exitCode).toBe(0);
+
+      // refresh subscription-manager
+      const subManRefresh = await client.Exec(['subscription-manager', 'refresh', '--force']);
+      expect(subManRefresh?.exitCode).toBe(0);
+
+      // clean cached metadata
+      const dnfCleanAll = await client.Exec(['dnf', 'clean', 'all']);
+      expect(dnfCleanAll?.exitCode).toBe(0);
+
+      // List errata the system is vulnerable to
+      const exist = await client.Exec([
+        'sh',
+        '-c',
+        'dnf updateinfo --list --all | grep RH | sort | tail -n 1',
+      ]);
+      if (exist?.exitCode != 0) {
+        console.log(exist?.stdout);
+        console.log(exist?.stderr);
+      }
+      expect(exist?.exitCode).toBe(0);
+      firstVersion = exist?.stdout?.toString();
+    });
+
+    await test.step('Update the template date to latest', async () => {
+      const rowTemplate = await getRowByNameOrUrl(page, templateName);
+      await rowTemplate.getByRole('button', { name: templateName }).click();
+      await expect(page.getByLabel('Breadcrumb').first()).toHaveText('RHELContentTemplates');
+      await expect(page.getByRole('heading', { level: 1 })).toHaveText(templateName);
+      await expect(page.getByText('Description:Template test')).toBeVisible();
```

Comment:
What do these 3 rows do? You are waiting for the page to load?

Reply:
Yes, but I think 3 is probably unneeded so I took two of them out